File: socketd/src/main/java/org/noear/socketd/protocol/ChannelAssistant.java
package org.noear.socketd.protocol;

import java.io.IOException;
import java.net.InetAddress;

/**
 * 通道助理
 *
 * @author noear
 * @since 2.0
 */
public interface ChannelAssistant<T> {
    /**
     * 写入
     *
     * @param target 目标
     * @param frame  帧
     */
    void write(T target, Frame frame) throws IOException;

    /**
     * 是否有效
     */
    boolean isValid(T target);

    /**
     * 关闭
     */
    void close(T target) throws IOException;

    /**
     * 获取远程地址
     */
    InetAddress getRemoteAddress(T target) throws IOException;

    /**
     * 获取本地地址
     */
    InetAddress getLocalAddress(T target) throws IOException;
}

File: swift/socketd/socketd/src/socketd/transport/core/ChannelAssistant.swift
//
//  ChannelAssistant.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

protocol ChannelAssistant{
    associatedtype T;
    
    /**
     * 写入
     *
     * @param target 目标
     * @param frame  帧
     */
    func write(_ target:T, _ frame:Frame);
    
    /**
     * 是否有效
     */
    func isValid(_ target:T) -> Bool;
    
    /**
     * 关闭
     */
    func close(_ target:T);
    
    /**
     * 获取远程地址
     */
    //func getRemoteAddress(_ target:T) -> InetSocketAddress;
    
    /**
     * 获取本地地址
     */
    //func getLocalAddress(_ target:T) -> InetSocketAddress;
}
